examples/quickstart-xgboost/client.py

```diff
@@ -0,0 +1,148 @@
+    "max_depth": 8,
+    "eval_metric": "auc",
+    "nthread": 16,
+    "num_parallel_tree": 1,
+    "subsample": 1,
+    "tree_method": "hist",
+}
+
+
+# Define Flower client
+class FlowerClient(fl.client.Client):
+    def __init__(self):
+        self.bst = None
+
+    def get_parameters(self, ins: GetParametersIns) -> GetParametersRes:
+        _ = (self, ins)
+        return GetParametersRes(
+            status=Status(
+                code=Code.OK,
+                message="OK",
+            ),
+            parameters=Parameters(tensor_type="", tensors=[]),
+        )
+
+    def local_boost(self):
+        # update trees based on local training data.
+        for i in range(num_local_round):
+            self.bst.update(train_dmatrix, self.bst.num_boosted_rounds())
+
+        # extract the last N=num_local_round trees as new local model
+        bst = self.bst[
+            self.bst.num_boosted_rounds()
+            - num_local_round : self.bst.num_boosted_rounds()
+        ]
+        return bst
+
+    def fit(self, ins: FitIns) -> FitRes:
+        if not self.bst:
+            # first round local training
+            print("Start training at round 1")
+            bst = xgb.train(
+                params,
+                train_dmatrix,
+                num_boost_round=num_local_round,
+                evals=[(valid_dmatrix, "validate"), (train_dmatrix, "train")],
+            )
+            self.config = bst.save_config()
+            self.bst = bst
+        else:
+            print("load global model")
+            for item in ins.parameters.tensors:
+                global_model = bytearray(item)
+
+            # load global model into booster
+            self.bst.load_model(global_model)
+            self.bst.load_config(self.config)
+
+            bst = self.local_boost()
+
+        local_model = bst.save_raw("json")
+        local_model_bytes = bytes(local_model)
+
+        return FitRes(
+            status=Status(
+                code=Code.OK,
+                message="OK",
+            ),
+            parameters=Parameters(tensor_type="", tensors=[local_model_bytes]),
+            num_examples=num_train,
+            metrics={},
+        )
+
+    def evaluate(self, ins: EvaluateIns) -> EvaluateRes:
+        eval_results = self.bst.eval_set(
+            evals=[(train_dmatrix, "train"), (valid_dmatrix, "valid")],
+            iteration=self.bst.num_boosted_rounds() - 1,
+        )
+        auc = round(float(eval_results.split("\t")[2].split(":")[1]), 4)
+
+        global_round = ins.config["global_round"]
+        print(f"AUC = {auc} at round {global_round}")
+
+        return EvaluateRes(
+            status=Status(
+                code=Code.OK,
+                message="OK",
+            ),
+            loss=0.0,
+            num_examples=num_val,
+            metrics={"AUC": auc},
+        )
+
+
+# Start Flower client
+fl.client.start_client(
+    server_address="127.0.0.1:8080", client=FlowerClient().to_client()
```

Comment:
The `to_client()` breaks my code. I think that was used for NumpyClient but I'm not sure how it should be used

Reply:
What error did you get?
Can you try this? `fl.client.start_client(server_address="127.0.0.1:8080", client=FlowerClient())`

Reply:
It works without `to_client()`. 
1. I'm curious why it works on your machine with the `to_client`.
2. By the way, the training took just ~1.5 seconds. Does it work that quickly for you, too?

Reply:
1. i used the one without `to_client()`, and got an msg suggesting me to add to_client(). it may due to the flwr version? i'm not sure. @danieljanes Do you have any idea?
2. yes it's quick as well on my side. i think it's due to the little amount of data.

Reply:
What version of Flower do you use? For me, it was 1.5

Reply:
i'm using 1.6.0

Reply:
1.6 is not released yet, you probably installed it via poetry. We should make it compatible with the available version, and also to make it work with the version that works according to the specification in requirements and pyproject.toml